onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -105,32 +106,32 @@ def create_attention_node(self, mask_index, q_matmul, k_matmul, v_matmul, q_add,
             return None
 
         qw = numpy_helper.to_array(q_weight)
-        assert qw.shape == (self.hidden_size, self.hidden_size)
+        assert qw.shape == (self.hidden_size, self.num_heads * self.head_size)
```

Comment:
we could still use hidden_size

Reply:
Thanks for the review @wangyems .
(self.hidden_size != self.num_heads * self.head_size) in case the attention heads are pruned.